### ISSUE TITLE
Thread safety

### ISSUE THREAD

**Issue by chastell** [troessner/reek#554]
We’re currently not sure whether `reek` is thread-safe, but strongly suspect it isn’t (at least as long as we use `Find.prune`, but most probably in other places as well).

We should investigate whether we can make it thread-safe (and whether it’s worth it in the first place).

**Comment by mvz** [comment]
I think this is worthwhile, being able to examine sources in separate threads could lead to big speed-ups on rubies with multi-processor support.

**Comment by troessner** [comment]
I agree. Let's fix this. But this kind of implies to go through our complete codebase and check all-the-things, right?.:)

**Comment by chastell** [comment]
Right, I don’t think there are any automated ways we could try to employ here… Did a cursory read of [this piece](https://bearmetal.eu/theden/how-do-i-know-whether-my-rails-app-is-thread-safe-or-not/) and it seems to have the right ideas.
